test/src/example.h

```diff
@@ -11,6 +11,10 @@
 // So we include it and make sure CI warns us if we use something that conflicts
 // with a Windows define.
 #ifdef WIN32
+// Make sure to disable min and max macros from the windows header.
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
```

Comment:
I would like to avoid adding this to cmake/scons because it's a user defined macro, and it could also be achieved by including the templates **before** `windows.h` (if your library really really need windows min/max definitions). WDYT @akien-mga ?

Reply:
I think this is defeating the purpose of this test including `windows.h` to see if it breaks stuff.
It does break stuff apparently, so we need to fix our includes to prevent this error. Defining `NOMINMAX` in user code is definitely a good idea for extensions using `windows.h`, but for this test specifically the aim is to make sure we can build without defining it.

We already replace or remove definitions of `MIN`/`MAX`/`CLAMP` in `include/godot_cpp/core/math.hpp` and `include/godot_cpp/core/defs.hpp`. Any code that uses `MIN` or `MAX` should likely be including our own header that defines them.

Reply:
I'm fine merging this solution for now to unblock this PR though, but we need to look into this further at some point. I'm also not clear on why it's starting to fail with this PR which isn't adding uses of `MIN` or `MAX` (which IMO should be replaced by `Math::min` and `Math::max`).

Reply:
> I'm also not clear on why it's starting to fail with this PR which isn't adding uses of `MIN` or `MAX` (which IMO should be replaced by `Math::min` and `Math::max`).

The problem is that `windows.h` without `NOMINMAX` defines `min` and `max`, not (just) `MIN` and `MAX`.
This causes even code like this to produce unexpected results:

```cpp
using std;
auto m = min(1, 2);
```

Reply:
An alternative is to:
```cpp
#ifdef min
#undef min
#endif
#ifdef max
#undef max
#endif
```

somewhere in the includes (I've tested in utility_functions, but could probably go somewhere else)

Reply:
oh, we do that in in `include/godot_cpp/core/defs.hpp` but since that gets included before `windows.h` it then gets re-defined.
There is no real solid solution since we don't know the order in which `windows.h` will be included.
The only alternative I found is to surround function calls (and definitions) with parenthesis, e.g.:

```cpp
(Math::min)(a, b)
```

But that would include user code, which will break on windows msvc when calling `Math::min(a, b)` instead of `(Math::min(a, b))`